src/flint/fmpz_mod_mat.jl

```diff
@@ -0,0 +1,863 @@
+          (Ref{T}, Ref{T}, Ref{T}), z, x, y)
+  return z
+end
+
+function -(x::T, y::T) where T <: Zmod_fmpz_mat
+  check_parent(x,y)
+  z = similar(x)
+  ccall((:fmpz_mod_mat_sub, :libflint), Nothing,
+          (Ref{T}, Ref{T}, Ref{T}), z, x, y)
+  return z
+end
+
+function *(x::T, y::T) where T <: Zmod_fmpz_mat
+  (base_ring(x) != base_ring(y)) && error("Base ring must be equal")
+  (ncols(x) != nrows(y)) && error("Dimensions are wrong")
+  z = similar(x, nrows(x), ncols(y))
+  ccall((:fmpz_mod_mat_mul, :libflint), Nothing,
+          (Ref{T}, Ref{T}, Ref{T}), z, x, y)
+  return z
+end
+
+
+################################################################################
+#
+#  Unsafe operations
+#
+################################################################################
+
+function mul!(a::T, b::T, c::T) where T <: Zmod_fmpz_mat
+  ccall((:fmpz_mod_mat_mul, :libflint), Nothing, (Ref{T}, Ref{T}, Ref{T}), a, b, c)
+  return a
+end
+
+function add!(a::T, b::T, c::T) where T <: Zmod_fmpz_mat
+  ccall((:fmpz_mod_mat_add, :libflint), Nothing, (Ref{T}, Ref{T}, Ref{T}), a, b, c)
+  return a
+end
+
+function zero!(a::T) where T <: Zmod_fmpz_mat
+  ccall((:fmpz_mod_mat_zero, :libflint), Nothing, (Ref{T}, ), a)
+  return a
+end
+
+################################################################################
+#
+#  Ad hoc binary operators
+#
+################################################################################
+
+function *(x::T, y::Int) where T <: Zmod_fmpz_mat
+  z = similar(x)
+  ccall((:fmpz_mod_mat_scalar_mul_si, :libflint), Nothing,
+          (Ref{T}, Ref{T}, Int), z, x, y)
+  return z
+end
+
+*(x::Int, y::T) where T <: Zmod_fmpz_mat = y*x
+
+function *(x::T, y::fmpz) where T <: Zmod_fmpz_mat
+  z = similar(x)
+  ccall((:fmpz_mod_mat_scalar_mul_fmpz, :libflint), Nothing,
+	(Ref{T}, Ref{T}, Ref{fmpz}), z, x, y)
+  return z
+end
+
+*(x::fmpz, y::T) where T <: Zmod_fmpz_mat = y*x
+
+function *(x::T, y::Integer) where T <: Zmod_fmpz_mat
+  return x*fmpz(y)
+end
+
+*(x::Integer, y::T) where T <: Zmod_fmpz_mat = y*x
+
+function *(x::fmpz_mod_mat, y::fmpz_mod)
+  (base_ring(x) != parent(y)) && error("Parent objects must coincide")
+  return x*y.data
+end
+
+*(x::fmpz_mod, y::fmpz_mod_mat) = y*x
+
+################################################################################
+#
+#  Powering
+#
+################################################################################
+
+#= Not implemented in Flint yet
+
+function ^(x::T, y::Int) where T <: Zmod_fmpz_mat
+  if y < 0
+     x = inv(x)
+     y = -y
+  end
+  z = similar(x)
+  ccall((:fmpz_mod_mat_pow, :libflint), Nothing,
+          (Ref{T}, Ref{T}, Int), z, x, y)
+  return z
+end
+=#
+
+function ^(x::T, y::fmpz) where T <: Zmod_fmpz_mat
+  (y > fmpz(typemax(Int))) &&
+          error("Exponent must be smaller then ", fmpz(typemax(Int)))
+  (y < fmpz(typemin(Int))) &&
+          error("Exponent must be bigger then ", fmpz(typemin(Int)))
```

Comment:
Is "then" an official valid substitute for "than" (just asking, I always wondered as I see this form so often). 
I suggest merging these two checks into one, e.g. 
```julia
typemin(Int) <= y <= typemax(Int) ||
    throw(DomainError(y, "exponent must be in $(typemin(Int):typemax(Int))"))
```

Reply:
Fixed.